lnwallet/channel.go

```diff
@@ -2334,133 +2343,280 @@ func NewBreachRetribution(chanState *channeldb.OpenChannel, stateNum uint64,
 		return nil, err
 	}
 
-	// In order to fully populate the breach retribution struct, we'll need
-	// to find the exact index of the commitment outputs.
-	ourOutpoint := wire.OutPoint{
-		Hash: commitHash,
-	}
-	theirOutpoint := wire.OutPoint{
-		Hash: commitHash,
+	theirDelay := uint32(chanState.RemoteChanCfg.CsvDelay)
+	theirScript, err := CommitScriptToSelf(
+		chanState.ChanType, isRemoteInitiator, keyRing.ToLocalKey,
+		keyRing.RevocationKey, theirDelay, leaseExpiry,
+	)
+	if err != nil {
+		return nil, err
 	}
-	for i, txOut := range revokedSnapshot.CommitTx.TxOut {
-		switch {
-		case bytes.Equal(txOut.PkScript, ourScript.PkScript):
-			ourOutpoint.Index = uint32(i)
-		case bytes.Equal(txOut.PkScript, theirScript.PkScript):
-			theirOutpoint.Index = uint32(i)
+
+	// Define an empty breach retribution that will be overwritten based on
+	// different version of the revocation log found.
+	br := &BreachRetribution{} // nolint: wastedassign
```

Comment:
Does using `new(BreachRetribution)` make that warning go away?

Reply:
Ah I see it now, but then we can just use a var statement here since the pointer gets overridden in either case?

Reply:
Cool changed.